providers/terraform-provider-csbpg/csbpg/connection_factory.go

```diff
@@ -28,9 +30,25 @@ func (c connectionFactory) Connect() (*sql.DB, error) {
 }
 
 func (c connectionFactory) uri() string {
-	return fmt.Sprintf(uriFormat, c.username, c.password, c.host, c.port, c.database)
+	return strings.Join([]string{
+		"host=" + c.host,
+		fmt.Sprintf("port=%d", c.port),
+		"user=" + c.username,
+		"password=" + c.password,
+		"database=" + c.database,
+		"sslmode=" + c.sslMode,
+		"sslinline=true",
+		fmt.Sprintf("sslcert='%s'", c.sslClientCert.Certificate),
+		fmt.Sprintf("sslkey='%s'", c.sslClientCert.Key),
+		fmt.Sprintf("sslrootcert='%s'", c.sslRootCert),
+	}, " ")
 }
 
 func (c connectionFactory) uriRedacted() string {
-	return fmt.Sprintf(uriFormat, c.username, "REDACTED", c.host, c.port, c.database)
+	return strings.ReplaceAll(c.uri(), c.password, "REDACTED")
```

Comment:
thoughts: `ReplaceAll` function calls `Replace` using -1 in the last parameter. It seems that 
`Replace(c.uri(), c.password, "REDACTED", 1)` would be more accurate since the password will only appear once.